roles/setup_lvm_lv/tasks/main.yml

```diff
@@ -0,0 +1,149 @@
+# The role default variable are not present in hostvars[inventory_hostname]
+# https://github.com/ansible/ansible/issues/6189
+# We then rely on "vars" which "works" but not happy about it
+- name: "LVM Setup: Check that VG specific variables are defined"
+  tags:
+    - assertion
+  assert:
+    that:
+      - vars['metalk8s_host_path_' ~ item.key] is defined
+  with_dict: '{{ metalk8s_lvm_managed_vgs }}'
+
+- name: 'LVM Setup: Check that you have mkfs for ext4 and xfs filesystem'
+  package:
+    name: '{{ item }}'
+    state: present
+  with_items:
+    - e2fsprogs
+    - xfsprogs
+
+- name: "LVM Setup: Gather fact with LVM data"
+  setup:
+    gather_subset: 'hardware'
+    filter: 'ansible_lvm'
+
+- name: "LVM Setup: Display LVs defined"
+  debug:
+    var: metalk8s_lvm_lvs
+  when: debug|bool
+
+- name: 'LVM Setup: Compute all LVM logical volumes'
+  set_fact:
+    # Combine te default properties with the one defined on a per LVM LV basis
+    metalk8s_lvm_all_lvs: >-
+      {
+        {%- set default_vg = metalk8s_lvm_default_vg|default("vg_metalk8s") -%}
+        {%- set device_prefix = "/dev/mapper/" -%}
+        {%- for lv_name, lv_prop in metalk8s_lvm_lvs.items() -%}
+          '{{ device_prefix }}{{ default_vg }}-{{ lv_name.replace("-", "--") }}':
+            {{ lv_prop|combine({
+              'lv_name': lv_name,
+              'vg_name': default_vg,
+              'force': lv_prop.force|default(metalk8s_lvm_lv_defaults_force),
+              'fs_opts': lv_prop.fs_opts|default(metalk8s_lvm_lv_defaults_fs_opts),
+              'fstype': lv_prop.fstype|default(metalk8s_lvm_lv_defaults_fstype),
+              'mount_opts': lv_prop.mount_opts|default(metalk8s_lvm_lv_defaults_mount_opts),
+              'labels': lv_prop.labels|default(metalk8s_lvm_lv_defaults_labels),
```

Comment:
If we specify label we loose the default label: `scality.com/metalk8s_lv: true`. Is that a wanted feature ?
(or it's a temporary hack to get label working at least basically, and then: never mind)

Reply:
No, should be merged.

Reply:
Also, not sure this should be `scality.com/metalk8s_lv: true`. What about `scality.com/metalk8s_vg: $VG_NAME`, then one can find all MetalK8s-managed PVs using a set-based label selector like `scality.com/metalk8s_vg` I believe, or obviously all those of a specific VG as well.

Similarly, setting `scality.com/metalk8s_node` to the node name could help as well.

Reply:
Good idea, will do; And yes indeed it was going to be merged.